core/src/avm2/object/script_object.rs

```diff
@@ -102,6 +102,17 @@ impl<'gc> ScriptObject<'gc> {
         ))
         .into()
     }
+
+    /// A special case for `newcatch` implementation. Basically a variable (q)name
+    /// which maps to slot 1.
+    pub fn catch_scope(mc: MutationContext<'gc, '_>, qname: &QName<'gc>) -> Object<'gc> {
+        let mut base = ScriptObjectData::custom_new(None, None);
+        let vt = VTable::newcatch(mc, &qname);
+        base.set_vtable(vt);
+        base.install_slot();
+        base.install_slot();
```

Comment:
Why do we call install slot twice here?

Reply:
Added comment to explain.

Reply:
Would this work?
```
pub fn catch_scope(activation: &mut ...
// ...
    base.install_instance_slots(activation);
}
// ...
     let so = ScriptObject::catch_scope(self, &qname);
```

Reply:
Done - base.install_instance_slots() doesn't even need activation.